lib/KeyWrapper.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+# include "Libconfig.h"
+# include <cstring>
+# include <memory>
+
+namespace LibConfig
+{
+  struct KeyWrapper
+  {
+    std::unique_ptr<char> data;
```

Comment:
I think you need to use `char[]` in this scenario, otherwise the destructor will call `delete key` instead of `delete [] key`
```suggestion
    std::unique_ptr<char[]> data;
```